docs/_articles/en/v3.x/salesforce-graph-engine/working-with-sfge.md

```diff
@@ -80,6 +80,36 @@ Example usage:
 public class MyClass {
 ```
 
+## "OutOfMemory: Java heap space" Error
+
+The number of paths SFGE needs to create increases exponentially with every conditional or method invocation in your code. As the number of paths explode, SFGE's probability of encountering an OutOfMemory error increases. In addition, a variety of factors such as your OS type, java setup, and other processes running in your machine can influence the heap space assigned by JVM, which then impacts SFGE's execution.
+
+Even if SFGE's execution was interrupted, it returns results from the portion of source code it has analyzed so far.
+
+While we are working on optimizing SFGE's heap consumption, here are some tips to avoid an OutOfMemory error:
+1. Increase max heap size assigned to SFGE's execution. You can do this by providing an updated `-Xmx` value to either the `sfgejvmargs` parameter with `scanner:run:dfa` command, or to the environment variable, `SFGE_JVM_ARGS`.
+
+	Before you do this, make note of your JVM's default heap size:
+	- Windows: `javaw -XX:+PrintFlagsFinal | find "MaxHeapSize"`
+	- Linux: `jstat -gccapacity <insert-sfge-pid-here>`
+	- Mac OS: `jps -v | grep <insert-sfge-pid-here>` and look for `-Xmx` value.
+
+	Execute SFGE with a larger heap space than the default settings. For example, to allocate 2g heap space, you can execute:
+	```bash
+	sfdx scanner:run:dfa --sfgejvmargs "-Xmx2g" <rest of your parameters>
+	```
+	or
+	```bash
+	export SFGE_JVM_ARGS="-Xmx2g"
+	sfdx scanner:run:dfa <rest of your parameters>
+	```
+
+	There's no single magic number that works for every project since the value depends on the project code's complexity. We recommend incrementing heap space allocation by at least 1g. Also, note that a very large heap space could slowdown the process further and degrade the performance. You may need a couple of tries to identify what works in your case.
```

Comment:
"since the value depends on the complexity of the target codebase."
"We recommend increasing heap space allocation in increments of 1g"
"slow down"

Reply:
"Since the value depends on the project code's complexity, there's no single magic number. We recommend that you increment heap space allocation by at least 1 G. A very large heap space could slowdown the process and degrade performance. To identify what works for your project, you may need to experiment."